implemented NOA filings + Colin-only filings

severinbeauvais opened the pull request:
*Issue #:* /bcgov/entity#4210

*Description of changes:*
This PR contain multiple commits that can be reviewed separately.

1. Pre-implementation Cleanup:
- renamed an enum mixin method
- renamed some interfaces for consistency
- removed unneeded enum declarations
- renamed todoItems and historyItems to distinguish them from filings
- misc cleanup

2a. Implementation:
- added ColinFiling component
- added NoticeOfAlteration component
- added history item subtitle
- added history item action button labels
- added NOA and Colin filing handling
- added types to filings, tasks, history items and task items
- added new entity names and types
- added new filing name and type
- updated enum mixin
- added Alteration interface
- added Filing interface
- added History Item interface
- added Task Item interface
- renamed interfaces consistently
- added/updated unit tests

2b. Improvements + Unit Tests:
- added Business interface
- added Header interface
- added unit tests for NOAs
- added unit test for Colin filings

2c. Text Updates per Janis/Kaine:
- updated NOA subtype labels
- updated NOA component text
- updated NOA title + mixin
- updated unit tests

2d. More Updates per Updated Schema:
- removed obsolete alteration object
- renamed alteration object per schema
- added alteration interfaces
- removed obsolete unit test
- updated unit tests

2e. More Updates and Improvements:
- updated expand button label per Janis
- added props to Header interface
- updated unit tests

2f. Preparation for New Entity Types (Corp Types):
- renamed EntityTypes -> LegalTypes (consistent with BE)
- renamed EntityNames -> LegalNames
- updated/added legal types and names consistent with Colin
- added NameRequestTypes to handle differences between Namex and Lear
- added "filed" label method with conditionals
- misc cleanup

2g. added missing semi-colons

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the business-filings-ui license (Apache 2.0).

lekshmimallika-aot commented on src/interfaces/alteration-interface.ts:
```diff
@@ -0,0 +1,12 @@
+  alterCorpType?: {
+    corpType: EntityTypes;
+  }
+  alterResolutions?: any;
```
There is no sub structure like alterResolutions. We have an array of resolution dates within alterShareStructure

severinbeauvais replied:
Fixed. Please review latest code.

lekshmimallika-aot commented on src/interfaces/alteration-interface.ts:
```diff
@@ -0,0 +1,12 @@
+  }
+  alterResolutions?: any;
+  alterCorpName?: any;
+  alterNameTranslation?: any;
```
alterNameTranslations

severinbeauvais replied:
Fixed. Please review latest code.

cameron-freshworks commented on src/interfaces/alteration-interface.ts:
```diff
@@ -0,0 +1,33 @@
+
+export interface AlterCorpTypeIF {
+  corpType: EntityTypes;
+  [propName: string]: any; // excess properties
```
This is pretty interesting!

cameron-freshworks commented on src/interfaces/business-interface.ts:
```diff
@@ -0,0 +1,9 @@
+export interface BusinessIF {
+  identifier: string,
+  legalName: string,
+  legalType?: FilingTypes,
```
Maybe missed some semicolons here.

severinbeauvais replied:
Fixed in next commit.

cameron-freshworks commented on src/interfaces/alteration-interface.ts:
```diff
@@ -0,0 +1,33 @@
+  alterCorpType?: AlterCorpTypeIF
+  alterCorpName?: AlterCorpNameIF
+  alterNameTranslations?: AlterNameTranslationsIF
+  alterShareStructure?: AlterShareStructureIF
```
Were the semicolons intentionally left out here?

severinbeauvais replied:
Fixed in next commit.

cameron-freshworks commented:
Not sure i understand the SonarCloud code warning.

severinbeauvais commented:
> Not sure i understand the SonarCloud code warning.

I looked into it. Apparently I caused 74 duplicated lines and 2 duplicate blocks -- both in the updated LegalNames / LegalTypes enums. Sure enough, these enums have the same set of constants, but different values assigned in each file. I wouldn't want to write this code any differently so I'm inclined to ignore this "failure" :)

cameron-freshworks commented:
> > Not sure i understand the SonarCloud code warning.
> 
> I looked into it. Apparently I caused 74 duplicated lines and 2 duplicate blocks -- both in the updated LegalNames / LegalTypes enums. Sure enough, these enums have the same set of constants, but different values assigned in each file. I wouldn't want to write this code any differently so I'm inclined to ignore this "failure" :)

Absolutely, no problems there!
The one in question i was referring too was suggesting renaming the 'NamexRequestMixin'.. but i didn't quite get why.

severinbeauvais commented:
> The one in question i was referring too was suggesting renaming the 'NamexRequestMixin'.. but i didn't quite get why.

Because the subject mixin has nothing to do with Namex. We get the NR data from legal-api, and we don't care what it connects to for the data (namex or nro or whatever). This mixin is purely working with a NR.

cameron-freshworks commented:
> > The one in question i was referring too was suggesting renaming the 'NamexRequestMixin'.. but i didn't quite get why.
> 
> Because the subject mixin has nothing to do with Namex. We get the NR data from legal-api, and we don't care what it connects to for the data (namex or nro or whatever). This mixin is purely for working with a NR.

I see, so maybe it is a valid warning!

severinbeauvais commented:
> > > The one in question i was referring too was suggesting renaming the 'NamexRequestMixin'.. but i didn't quite get why.
> > 
> > 
> > Because the subject mixin has nothing to do with Namex. We get the NR data from legal-api, and we don't care what it connects to for the data (namex or nro or whatever). This mixin is purely for working with a NR.
> 
> I see, so maybe it is a valid warning!

For other reviewers... I got this all wrong. Cam was asking about the code smell that SonarCloud reported. It was because I renamed the subject mixin file but I forgot to rename the exported class! (SC just recommended that filename and export should match.) So I've now fixed this.
